src/lib/y2firewall/clients/installation_finish.rb

```diff
@@ -54,9 +54,15 @@ def modes
 
       def write
         Service.Enable("sshd") if @settings.enable_sshd
+        configure_firewall if @firewalld.installed?
+        true
```

Comment:
i think that methods which always returns ```true``` can easily be transformed into return ```nil```. No one will check the return value anyway.

On the other hand should it really be always true?

Reply:
It will enable the ssh service and will configure the firewall according to the proposal. Once done it will return true, if something fails with the firewalld configuration, it will be reported by the firewalld command line, except for the support of the service that will be logged.

I decided to return true once the client did its job, in case in the future we would like to return false in some condition we can add it safety.

And it is a common return implementation over finish clients:

https://github.com/yast/yast-bootloader/blob/master/src/lib/bootloader/finish_client.rb#L73

Reply:
yes, I can read the code.

The point should be that it is already masking return value of ```Service.Enable```. And returning true when firewall could not be configured is arguable but acceptable in the finish client.

Reply:
If I'm not wrong in other clients it was not a condition of return. I mean, we tried to enable the service failing silently and it behaves exactly as it was.

https://github.com/yast/yast-network/blob/SLE-12-SP1/src/clients/firewall_stage1_finish.rb#L68

Reply:
I think you are both missing an important point: you are not designing an API. You are implementing an instance of the [FinishClient](http://www.rubydoc.info/github/yast/yast-yast2/Installation/FinishClient) API. So the **caller** decides how success and failure is communicated.

(Unfortunately, [the API docs is silent on this point](https://github.com/yast/yast-yast2/blob/f829c95f3f284fff9ee93a77d3278fb2ecffe942/library/general/src/lib/installation/finish_client.rb#L66))

Reply:
Well, it maybe came from ycp times but that extra checks are usually not done:

https://github.com/yast/yast-storage-ng/blob/master/src/lib/y2storage/clients/finish.rb

Reply:
Maybe something to document? Best practices for installation finish clients?